public/app/features/alerting/unified/components/DynamicTable.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { FC, ReactNode } from 'react';
+import { css, cx } from '@emotion/css';
+import { GrafanaTheme } from '@grafana/data';
+import { IconButton, useStyles, useTheme } from '@grafana/ui';
+import { useMedia } from 'react-use';
+
+export interface DynamicTableColumnProps<T = any> {
+  id: string | number;
+  label: string;
+
+  renderRow?: (item: DynamicTableItemProps<T>) => ReactNode;
+  size?: number | string;
+}
+
+export interface DynamicTableItemProps<T = any> {
+  id: string | number;
+  data: T;
+
+  renderExpandedContent?: () => ReactNode;
+  isExpanded?: boolean;
+}
+
+export interface DynamicTableProps<T = any> {
+  cols: Array<DynamicTableColumnProps<T>>;
```

Comment:
nit: any particular reason why we use `cols` instead of `columns`?

Reply:
That's how i'm used to shorten `columns` 😄